source/plugins/go/src/oms.go

```diff
@@ -1149,6 +1152,141 @@ func UpdateNumTelegrafMetricsSentTelemetry(numMetricsSent int, numSendErrors int
 	ContainerLogTelemetryMutex.Unlock()
 }
 
+func processIncludes(kubernetesMetadataMap map[string]interface{}, includesList []string) map[string]interface{} {
+	includedMetadata := make(map[string]interface{})
+
+	// pre process image related fields
+	var imageRepo, imageName, imageTag, imageID string
+	imageProcessed := false // Flag to check if image processing is required
+	for _, include := range includesList {
+		if include == "imageid" || include == "imagerepo" || include == "image" || include == "imagetag" {
+			imageProcessed = true
+			break
+		}
+	}
+
+	if imageProcessed {
+		if hash, ok := kubernetesMetadataMap["container_hash"].(string); ok {
+			imageID = extractImageID(hash)
+		}
+		if image, ok := kubernetesMetadataMap["container_image"].(string); ok {
+			imageRepo, imageName, imageTag = parseImageDetails(image)
+		}
+	}
+
+	for _, include := range includesList {
+		switch include {
+		case "poduid":
+			if val, ok := kubernetesMetadataMap["pod_id"]; ok {
+				includedMetadata["podUid"] = val
+			}
+		case "podlabels":
+			if val, ok := kubernetesMetadataMap["labels"]; ok {
+				includedMetadata["podLabels"] = val
+			}
+		case "podannotations":
+			if val, ok := kubernetesMetadataMap["annotations"]; ok {
+				if annotationsMap, ok := val.(map[string]interface{}); ok {
+					filteredAnnotations := make(map[string]interface{})
+					for key, annotationValue := range annotationsMap {
+						if !strings.Contains(key, "kubernetes.io/config") {
+							filteredAnnotations[key] = annotationValue
+						}
+					}
+					includedMetadata["podAnnotations"] = filteredAnnotations
```

Comment:
similarly for this as well, Did you test what max annotation size can be ingested? is it 64KB? if yes, then we should cap up to  64KB in the agent  and then discard annotations which are more than 64KB.

Reply:
There is no strict character limit for individual keys or values in annotations.

Reply:
![image](https://github.com/microsoft/Docker-Provider/assets/25573844/56608d7c-ed6f-4460-a79a-0a0a73ab6738)